.github/workflows/rust-test.yml

```diff
@@ -20,4 +20,4 @@ jobs:
         run:            git submodule update --init --recursive
       - name:           Rust build
         shell:          bash
-        run:            cd rust && cargo test && cd ..
+        run:            cd rust && cp os-specific-lock/android/Cargo.lock . && cargo test && cp os-specific-lock/ios/Cargo.lock . && cargo test && cd ..
```

Comment:
```suggestion
        run: |
          cd rust
          cp os-specific-lock/android/Cargo.lock .
          cargo test
          cp os-specific-lock/ios/Cargo.lock .
          cargo test
```

Reply:
or maybe even divide them by two tests that would run in parallel

Reply:
Done; now ios test is running on macos as ios will be built on that arch. Unfortunately, this flakes all tests with disk access operations. If I don't solve this soon, I'll move ios tests back to linux, but we should do something about this in longer term.